File: Full_Stack_Exam_Portal/backendexam/src/main/java/com/exam/backendexam/model/Quiz.java
package com.exam.backendexam.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.util.HashSet;
import java.util.Set;

@Getter
@Setter
@NoArgsConstructor
@Entity
public class Quiz {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String title;
    private String description;

    private String maxMarks;
    private String numberOfQuestions;
    @Column(columnDefinition = "tinyint(1) default 0")
    private Boolean isActive;

    @ManyToOne(fetch = FetchType.EAGER)
    private Category category;


    @OneToMany(mappedBy = "quiz", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    @JsonIgnore
    private Set<Question> questions = new HashSet<>();
}

File: Full_Stack_Exam_Portal/front-end/examportal/src/app/quiz.ts
import { Category } from "./category";

export class Quiz {
  id: number = 0 ;
  title: string = '';
  description: string= '';
  maxMarks: string= '';
  numberOfQuestions: string= '';
  isActive: boolean= false;
  category : Category = new Category();
  quizLogo : string = '';
}